launcher/image/preload.sh

```diff
@@ -126,6 +128,36 @@ get_cos_gpu_installer_image_digest() {
   echo "${image_digest}"
 }
 
+validate_digest() {
+  driver_digest="${1}"
+  # Check for the expected length of the SHA256 digest (64 hex characters)
+  if [ ${#driver_digest} -ne 64 ]; then
+    echo "Error: driver digest has an unexpected length: ${#driver_digest}, Expected 64." >&2
+    return 1
+  fi
+  # Check for valid hexadecimal string
+  if [[ ! ${driver_digest} =~ ^[0-9a-fA-F]+$ ]]; then
+    return "Error: driver digest ${driver_digest} is not valid hexadecimal string." >&2
+    return 1
+  fi
+}
+
+set_driver_digest_reference() {
+  local driver_version
+  local driver_digest_gcs_url
+
+  # Fetching the default driver version for H100 GPU.
+  driver_version=$(cos-extensions list -- --target-gpu NVIDIA_H100_80GB | grep DEFAULT | cut -d" " -f 1)
```

Comment:
this func should take a parameter for the target gpu rather than hardcoding the H100